dask/tests/test_distributed.py

```diff
@@ -293,6 +302,10 @@ def reliable_double(x):
         "full",
     ],
 )
+@pytest.mark.filterwarnings(
+    "ignore:Passing a local execution scheduler in Dask.distributed might "
+    "lead to unexpected results."
+)
```

Comment:
I don't see where the scheduler is getting set explicitly in this test.

Reply:
Oof I was going to say maybe we can set the scheduler in the `assert_eq` call instead, but that does not work very well. The `"sync"` scheduler is still hardcoded in some of the nested functions and if you switch them to take scheduler as an input then you run into other issues. So I think this is fine as you have it for now, and we can open a separate ticket to make the `assert_eq` work better with distritbuted client.

Reply:
Oh! I forgot to comment. I merged a change in https://github.com/dask/dask/pull/8755 that fixes the hard coding. So I think you can get rid of some of these warning filters. 

> I see that you triggered the tests. In this case, should I still update the wording of the warning message as you suggested? Thanks!

Yes please update the wording

Reply:
Cool, just pushed the fixes. Btw, do you know if there's a more elegant way to deal with the warning messages? Right now we need to "hard-code" the messages everywhere we need it (in the filters, for example). I'm just curious about it!